src/ui_fsmenu/crash_report.cc

```diff
@@ -0,0 +1,155 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "ui_fsmenu/crash_report.h"
+
+#include "graphic/text_layout.h"
+#include "io/fileread.h"
+#include "ui_basic/box.h"
+#include "ui_basic/button.h"
+#include "ui_basic/multilinetextarea.h"
+#include "ui_basic/textarea.h"
+#include "ui_fsmenu/tech_info.h"
+
+namespace FsMenu {
+
+constexpr int kButtonSize = 100;
+
+CrashReportWindow::CrashReportWindow(FsMenu::MainMenu& menu, const FilenameSet& crash_files)
+   : UI::Window(&menu, UI::WindowStyle::kFsMenu, "crash", 0, 0, 0, 0, _("Crash Report")),
+     clear_on_close_(crash_files) {
+
+	UI::Box* box = new UI::Box(
+	   this, UI::PanelStyle::kFsMenu, "main_box", 0, 0, UI::Box::Vertical, 0, 0, kPadding);
+
+	// Only for the report. TODO(Nordfriese): Just pull the report assembly out into a new function.
+	TechInfoBox tech_info(box, TechInfoBox::Type::kAbout);
+	tech_info.set_visible(false);
+
+	// not localized
+	std::string detailed_text =
+	   format("# Autogenerated crash report for %u crash(es):\n\n", crash_files.size());
+	for (const std::string& filename : crash_files) {
+		detailed_text += "\n- ";
+		detailed_text += FileSystem::fs_filename(filename.c_str());
+	}
+	detailed_text += "\n\n## Technical report:\n";
+	detailed_text += tech_info.get_report();
+	for (const std::string& filename : crash_files) {
+		detailed_text += "\n\n## ";
+		detailed_text += filename;
+		detailed_text += "\n\n";
+
+		FileRead fr;
+		try {
+			fr.open(*g_fs, filename);
+			detailed_text += fr.data(fr.get_size());
+		} catch (const std::exception& e) {
+			detailed_text += "Error opening crash report file:\n";
+			detailed_text += e.what();
+		}
+	}
+
+	UI::MultilineTextarea* header = new UI::MultilineTextarea(
+	   box, "header", 0, 0, 200, 200, UI::PanelStyle::kFsMenu,
+	   as_richtext_paragraph(
+	      format(_("It looks like Widelands crashed the last time you ran it, but a backtrace for "
+	               "debugging was automatically generated. Please provide the report to the "
+	               "Widelands Development Team under %s so we can fix the problem."),
+	             as_url_hyperlink(kReportBugsURL)),
+	      UI::FontStyle::kFsMenuLabel),
+	   UI::Align::kCenter, UI::MultilineTextarea::ScrollMode::kNoScrolling);
+
+	std::string detailed_text_rt = richtext_escape(detailed_text);
+	newlines_to_richtext(detailed_text_rt);
+	detailed_text_rt =
+	   as_richtext_paragraph(detailed_text_rt, UI::FontStyle::kFsMenuInfoPanelParagraph);
+	UI::MultilineTextarea* details = new UI::MultilineTextarea(
+	   box, "details", 0, 0, 650, 150, UI::PanelStyle::kFsMenu, detailed_text_rt);
```

Comment:
This could be a little taller, e.g.
```suggestion
	   box, "details", 0, 0, 650, 250, UI::PanelStyle::kFsMenu, detailed_text_rt);
```

Reply:
I think since the crash report window is mainly aimed at less tech-savvy users, the highly technical, scary-looking details should not be displayed too prominently as this might make people close the window as "something technical" without reading the header

Reply:
That's a valid point. Would it be possible then to make it collapsible? So it wouldn't even be shown by default, but if someone wants to see what it is, it would be more comfortable.